crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -28,21 +29,24 @@ use std::collections::HashSet;
 pub fn execute_call_entry_point(
     entry_point: &mut CallEntryPoint, // Instead of 'self'
     state: &mut dyn State,
-    cheatnet_state: &mut CheatnetState, // Added parameter
+    runtime_state: &mut RuntimeState,
     resources: &mut ExecutionResources,
     context: &mut EntryPointExecutionContext,
 ) -> EntryPointExecutionResult<CallInfo> {
     // region: Modified blockifier code
     // We skip recursion depth validation here.
 
-    cheatnet_state
+    runtime_state
+        .cheatnet_state
         .trace_data
-        .enter_nested_call(entry_point.clone());
+        .enter_nested_call(entry_point.clone()); // TODO this should be handled by trace extension
 
-    if let Some(ret_data) = get_ret_data_by_call_entry_point(entry_point, cheatnet_state) {
+    if let Some(ret_data) =
+        get_ret_data_by_call_entry_point(entry_point, runtime_state.cheatnet_state)
+    {
         return Ok(mocked_call_info(entry_point.clone(), ret_data.clone()));
-    }
-    // endregion
+    } // TODO this should be removed from here, it should be handled by cheatable starknet extension
```

Comment:
Create an issue for it and tag it here please

Reply:
https://github.com/orgs/foundry-rs/projects/3/views/2?pane=issue&itemId=51750365